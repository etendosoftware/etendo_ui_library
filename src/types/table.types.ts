import { DimensionValue } from 'react-native';

export type Metadata = {
  primary?: boolean;
  visible?: boolean;
  label?: string;
  key?: string;
  type?: 'number' | 'string' | 'boolean' | 'date';
  width?: DimensionValue;
  title?: boolean;
};
