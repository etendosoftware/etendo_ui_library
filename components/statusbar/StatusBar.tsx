import React from 'react';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import {RED} from '../../styles/colors';
import {StatusBarStyleVariant} from './StatusBar.styles';
import {StatusBarProps} from './StatusBar.types';

export const StatusBar = ({data, step}: StatusBarProps) => {
  if (step === 1) {
    return (
      <View style={StatusBarStyleVariant.primary.containerStatusBar}>
        <View
          style={StatusBarStyleVariant.primary.containerStatusBarMiddleBlueTop}
        />
        <View
          style={
            StatusBarStyleVariant.primary.containerStatusBarMiddleYellowBottom
          }
        />

        <View style={StatusBarStyleVariant.primary.containerItemActive}>
          <Text style={StatusBarStyleVariant.primary.textActive}>
            {data[0].step}
          </Text>
        </View>
        <View style={StatusBarStyleVariant.primary.containerItemDisabled}>
          <Text
            style={StatusBarStyleVariant.primary.containerItemDisabledStep2Text}
          >
            {data[1].step}
          </Text>
          <Text
            style={StatusBarStyleVariant.primary.containerItemDisabledStep3Text}
          >
            {data[2].step}
          </Text>
        </View>
      </View>
    );
  } else if (step === 2) {
    return (
      <View style={StatusBarStyleVariant.primary.containerStatusBar}>
        <View
          style={StatusBarStyleVariant.primary.containerStatusBarMiddleBlueTop}
        />
        <View
          style={
            StatusBarStyleVariant.primary.containerStatusBarMiddleYellowBottom
          }
        />

        <View
          style={
            StatusBarStyleVariant.primary
              .containerStatusBarMiddleOfMiddleHorizontalYellowTop
          }
        />

        <View
          style={
            StatusBarStyleVariant.primary
              .containerStatusBarMiddleOfMiddleHorizontalBlueBottom
          }
        />

        <View
          style={
            StatusBarStyleVariant.primary
              .containerStatusBarMiddleOfMiddleHorizontalBlueTop
          }
        />

        <View style={StatusBarStyleVariant.primary.containerItemDisabledStep2}>
          <Text
            style={StatusBarStyleVariant.primary.containerItemDisabledStep1Text}
          >
            {data[0].step}
          </Text>
        </View>
        <View style={StatusBarStyleVariant.primary.containerItemActiveStep2}>
          <View
            style={StatusBarStyleVariant.primary.activeItemStep2LeftSection}
          >
            <Image
              source={data[1].activeImage}
              style={StatusBarStyleVariant.primary.step2Image}
            />
            <View style={StatusBarStyleVariant.primary.activeItemStep2Text}>
              <Text
                style={StatusBarStyleVariant.primary.activeItemStep2TextTop}
              >
                {data[1].activeTextTop}
              </Text>
              <Text
                style={StatusBarStyleVariant.primary.activeItemStep2TextBottom}
              >
                {data[1].activeTextBottom}
              </Text>
            </View>
          </View>
          <Text style={StatusBarStyleVariant.primary.activeItemStep2TextRight}>
            {data[1].step}
          </Text>
        </View>
        <View
          style={StatusBarStyleVariant.primary.containerItemDisabledStep3Right}
        >
          <Text
            style={
              StatusBarStyleVariant.primary.containerItemDisabledStep3TextRight
            }
          >
            {data[2].step}
          </Text>
        </View>
      </View>
    );
  } else if (step === 3) {
    return (
      <View style={StatusBarStyleVariant.primary.containerStatusBar}>
        <View
          style={
            StatusBarStyleVariant.primary.containerStatusBarMiddleYellowBottom
          }
        />
        <View
          style={StatusBarStyleVariant.primary.containerStatusBarMiddleBlueTop}
        />

        <View
          style={
            StatusBarStyleVariant.primary.containerItemDisabledStep1AndStep2
          }
        >
          <Text
            style={
              StatusBarStyleVariant.primary
                .containerItemDisabledStep1TextSecundary
            }
          >
            {data[0].step}
          </Text>
          <Text
            style={
              StatusBarStyleVariant.primary
                .containerItemDisabledStep2TextSecundary
            }
          >
            {data[1].step}
          </Text>
        </View>
        <View style={StatusBarStyleVariant.primary.containerItemActiveStep3}>
          <Image
            source={require('../../assets/images/icons/calendar-secundary.png')}
            style={StatusBarStyleVariant.primary.step3Image}
          />
          <View
            style={
              StatusBarStyleVariant.primary.containerItemActiveStep3LeftSection
            }
          >
            <Text style={StatusBarStyleVariant.primary.activeItemStep3TextTop}>
              {data[2].activeTextTop}
            </Text>

            <Text
              style={StatusBarStyleVariant.primary.activeItemStep3TextBottom}
            >
              {data[2].activeTextBottom}
            </Text>
          </View>

          <Text style={StatusBarStyleVariant.primary.textActiveStep3}>
            {data[2].step}
          </Text>
        </View>
      </View>
    );
  } else {
    return (
      <View style={StatusBarStyleVariant.primary.containerStatusBar}>
        <View
          style={StatusBarStyleVariant.primary.containerStatusBarMiddleBlueTop}
        />
        <View
          style={
            StatusBarStyleVariant.primary.containerStatusBarMiddleYellowBottom
          }
        />

        <View style={StatusBarStyleVariant.primary.containerItemActive}>
          <Text style={StatusBarStyleVariant.primary.textActive}>
            {data[0].step}
          </Text>
        </View>
        <View style={StatusBarStyleVariant.primary.containerItemDisabled}>
          <Text
            style={StatusBarStyleVariant.primary.containerItemDisabledStep2Text}
          >
            {data[1].step}
          </Text>
          <Text
            style={StatusBarStyleVariant.primary.containerItemDisabledStep3Text}
          >
            {data[2].step}
          </Text>
        </View>
      </View>
    );
  }
};
