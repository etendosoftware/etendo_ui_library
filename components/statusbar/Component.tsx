import React from 'react';
import {Image, Text, View} from 'react-native';
import {StatusBarStyleVariant} from './StatusBar.styles';

export const Component = () => {
  return (
    <>
      <Image
        source={require('../../assets/images/icons/calendar-secundary.png')}
        style={StatusBarStyleVariant.primary.activeComponentImage}
      />
      <View style={StatusBarStyleVariant.primary.activeStepContentRightSection}>
        <Text style={StatusBarStyleVariant.primary.textTopActiveStepContent}>
          Current Plan date
        </Text>
        <Text style={StatusBarStyleVariant.primary.textBottomActiveStepContent}>
          03 January, 2022
        </Text>
      </View>
    </>
  );
};
