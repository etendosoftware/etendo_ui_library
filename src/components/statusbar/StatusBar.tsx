/* Imports */
import React from 'react';
import { ViewEtendo } from '../viewEtendo/ViewEtendo';
import { StatusBarStyleVariant } from './StatusBar.styles';
import { StatusBarProps } from './StatusBar.types';
import { StepActive } from './StepActive';
import { StepDisabled } from './StepDisabled';

const StatusBar = ({ data, step }: StatusBarProps) => {
  return (
    <ViewEtendo style={StatusBarStyleVariant.primary.statusBarContainer}>
      {data.map((item, idx) => {
        if (idx === step - 1) {
          return <StepActive item={item} step={step} key={'status1' + idx} />;
        } else {
          return <StepDisabled item={item} step={step} key={'status2' + idx} />;
        }
      })}
    </ViewEtendo>
  );
};

export default StatusBar;
