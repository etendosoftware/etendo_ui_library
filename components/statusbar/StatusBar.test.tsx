import React from 'react';
import renderer from 'react-test-renderer';
import {Component} from './Component';

import {StatusBar} from './StatusBar';
import {Info} from './StatusBar.types';

const dataset: Info[] = [
  {
    step: '1. Vehicles',
    routeNav: '/vehicles',
    key: 'vehicles',
    componentActive: <Component />,
  },
  {
    step: '2. Visits',
    routeNav: '/visits',
    key: 'visits',
    componentActive: <Component />,
  },
  {
    step: '3. Routes',
    routeNav: '/routes',
    key: 'routes',
    componentActive: <Component />,
  },
  {
    step: '4. Capacity',
    routeNav: '/capacity',
    key: 'capacity',
    componentActive: <Component />,
  },
  {
    step: '5. Settings',
    routeNav: '/settings',
    key: 'settings',
    componentActive: <Component />,
  },
];

describe('Running Test for StatusBar', () => {
  it('Check StatusBar', () => {
    const tree = renderer
      .create(<StatusBar data={dataset} step={1} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
