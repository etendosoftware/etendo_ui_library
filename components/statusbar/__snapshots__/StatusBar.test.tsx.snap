// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running Test for StatusBar Check StatusBar Disabled 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#202452",
      "display": "flex",
      "flex": 1,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "flex-end",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#202452",
        "borderBottomLeftRadius": 20,
        "borderBottomStartRadius": 20,
        "borderTopRightRadius": 20,
        "display": "flex",
        "flex": 0.15,
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "600",
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      1. Vehicles
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#202452",
        "borderBottomLeftRadius": 20,
        "borderBottomStartRadius": 20,
        "borderTopRightRadius": 20,
        "display": "flex",
        "flex": 0.15,
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "600",
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      2. Visits
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#202452",
        "borderBottomLeftRadius": 20,
        "borderBottomStartRadius": 20,
        "borderTopRightRadius": 20,
        "display": "flex",
        "flex": 0.15,
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "600",
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      3. Routes
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#202452",
        "borderBottomLeftRadius": 20,
        "borderBottomStartRadius": 20,
        "borderTopRightRadius": 20,
        "display": "flex",
        "flex": 0.15,
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "600",
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      4. Capacity
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#202452",
        "borderBottomLeftRadius": 20,
        "borderBottomStartRadius": 20,
        "borderTopRightRadius": 20,
        "display": "flex",
        "flex": 0.15,
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "600",
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      5. Settings
    </Text>
  </View>
</View>
`;
